### PR TITLE
Revert disabling the toolset package in Helix

### PR DESCRIPTION
**Pull request by jjonescz**
Resolves https://github.com/dotnet/sdk/issues/42845.

### CODE REVIEW

**Comment by jjonescz** [comment]
@marcpopMSFT @jaredpar @baronfel for reviews, thanks

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 1]
```diff
@@ -232,6 +232,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(BuildWithNetFrameworkHostedCompiler)' == 'true' and '$(OS)' == 'Windows_NT'">
     <RoslynTargetsPath>$(NuGetPackageRoot)\microsoft.net.sdk.compilers.toolset\$(NETCoreSdkVersion)</RoslynTargetsPath>
     <_NeedToDownloadMicrosoftNetSdkCompilersToolsetPackage>true</_NeedToDownloadMicrosoftNetSdkCompilersToolsetPackage>
+    <_SkipCheckMicrosoftNetSdkCompilersToolsetPackageExists Condition="'$(NuGetPackageRoot)' == ''">true</_SkipCheckMicrosoftNetSdkCompilersToolsetPackageExists>
```
Comment (baronfel) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:235]: Can you add a comment/link to the issues/discussions that triggered this check? I expect future-us to forget and having some record of why this was necessary would be helpful.

File: build/RunTestsOnHelix.cmd [hunk 2]
```diff
@@ -38,6 +37,7 @@ REM We downloaded a special zip of files to the .nuget folder so add that as a s
 dotnet nuget list source --configfile %TestExecutionDirectory%\nuget.config
 PowerShell -ExecutionPolicy ByPass "dotnet nuget locals all -l | ForEach-Object { $_.Split(' ')[1]} | Where-Object{$_ -like '*cache'} | Get-ChildItem -Recurse -File -Filter '*.dat' | Measure"
 dotnet nuget add source %DOTNET_ROOT%\.nuget --configfile %TestExecutionDirectory%\nuget.config
+if exist %TestExecutionDirectory%\Testpackages dotnet nuget add source %TestExecutionDirectory%\Testpackages --name testpackages --configfile %TestExecutionDirectory%\nuget.config
```
Comment (marcpopMSFT) [inline build/RunTestsOnHelix.cmd:40]: Does the script work as is for local helix execution or are changes needed to the instructions here: https://github.com/dotnet/sdk/blob/main/documentation/project-docs/repro-helix-failure.md

We don't do this often but that means it's fragile to changes if we're not careful.

Comment (jjonescz) [reply build/RunTestsOnHelix.cmd:40]: Didn't know about this possibility to repro helix locally, thanks.

I've followed the instructions and they seem to work, in particular the nuget source is added correctly:

```
C:\temp\helix\payload-dir>if exist C:\Users\janjones\AppData\Local\Temp\dotnetSdkTests\jxi5tacr.5hi\Testpackages dotnet nuget add source C:\Users\janjones\AppData\Local\Temp\dotnetSdkTests\jxi5tacr.5hi\Testpackages --name testpackages --configfile C:\Users\janjones\AppData\Local\Temp\dotnetSdkTests\jxi5tacr.5hi\nuget.config
Package source with Name: testpackages added successfully.
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets
Comment (jjonescz) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:1]: btw, this PR is targeting `release/9.0.1xx`, let me know if that's not correct

**Review by baronfel — APPROVED** [review]

**Comment by jjonescz** [comment]
@marcpopMSFT if this looks good, feel free to merge, thanks

**Comment by marcpopMSFT** [comment]
Looks like the local helix repro steps work with this change so we should be good here (that was my main concern).

**Pull request merged**